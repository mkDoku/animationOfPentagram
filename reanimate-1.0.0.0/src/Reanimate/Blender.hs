{-# LANGUAGE OverloadedStrings #-}
{-|
  [Blender](https://www.blender.org/) is a free and open-source 3D graphics toolkit.
  It is usually used through a graphical user-interface but can also be
  scripted via Python. These Blender scripts can access 100% of Blender's
  functionality and offer a convenient way of coding 3D effects.

  Running Blender can be time-consuming but heavy caching means scripts
  are only run when they change.

  Blender cheatsheet:

> # To generate with a transparent background, set film_transparent = True:
> bpy.context.scene.render.film_transparent = True
>
> # Filmic is great for photorealism but bad for animations.
> # If you want your textures to keep their exact color values,
> # set the view_transform to 'Standard':
> bpy.context.scene.view_settings.view_transform = 'Standard'
>
> # Blender's default render engine is 'EEVEE', fast but not a raytracer.
> # To switch to raytracing, set the engine to 'CYCLES':
> bpy.context.scene.render.engine = 'CYCLES'
>
> # Rendering at full resolution can be slow. When developing, try
> # decreasing the resolution_percentage for faster renders.
> bpy.context.scene.render.resolution_percentage = 10
>
> # The resolution of the final image are set by resolution_x and resolution_y:
> bpy.context.scene.render.resolution_x = 320
> bpy.context.scene.render.resolution_y = 180

-}
module Reanimate.Blender
  ( blender
  , blender'
  ) where

import           Data.Hashable
import           Data.Text                  (Text)
import qualified Data.Text.IO               as T
import           Graphics.SvgTree           (Tree)
import           Reanimate.Animation
import           Reanimate.Cache
import           Reanimate.Constants
import           Reanimate.Misc
import           Reanimate.Parameters
import           Reanimate.Raster
import           Reanimate.Svg.Constructors
import           System.FilePath            (replaceExtension, (<.>))
import           System.IO.Unsafe           (unsafePerformIO)

-- | Run a Blender script and embed the resulting image file. The
--   image will be scaled to fit the screen exactly (assuming a default
--   canvas layout). Note that Blender resolution defaults to 1920x1080
--   but can be changed in the script code.
blender :: Text -> SVG
blender script =
  unsafePerformIO $ mkBlenderImage script

-- | Generate Blender image as a separate PNG file. Can be embedded with
--   `mkImage`.
blender' :: Text -> FilePath
blender' script =
  unsafePerformIO $ mkBlenderImage' script

mkBlenderImage :: Text -> IO Tree
mkBlenderImage script | pNoExternals = pure $ mkText script
mkBlenderImage script =
  mkImage screenWidth screenHeight <$> mkBlenderImage' script

mkBlenderImage' :: Text -> IO FilePath
mkBlenderImage' _ | pNoExternals = pure "/blender/has/been/disabled"
mkBlenderImage' script = cacheFile template $ \target -> do
    exec <- requireExecutable "blender"
    let py_file = replaceExtension target "py"
    T.writeFile py_file script
    runCmd exec [ "--background","--render-format", "PNG"
                , "--python-exit-code", "1"
                , "--render-output", target, "--python", py_file]
  where
    template = encodeInt (hash script) <.> "png"
