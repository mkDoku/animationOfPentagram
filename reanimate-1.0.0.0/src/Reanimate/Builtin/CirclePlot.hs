{-|
Module      : Reanimate.Builtin.CirclePlot
Copyright   : Written by David Himmelstrup
License     : Unlicense
Maintainer  : lemmih@gmail.com
Stability   : experimental
Portability : POSIX

Convenience module for rendering circle plots.

-}
module Reanimate.Builtin.CirclePlot where

import           Codec.Picture
import           Graphics.SvgTree (Tree)
import           Reanimate.Raster
import           Reanimate.Svg
import           Reanimate.Constants

-- | Circle plots are scaled to 'screenHeight'.
--
--   Example:
--
-- @
-- 'circlePlot' 500 $ \\ang r ->
--   'Codec.Picture.Types.promotePixel' $ toRGB8 $ uncurryRGB sRGB $ hsv (ang/pi*180) r 1
-- @
--
--   <<docs/gifs/doc_circlePlot.gif>>
circlePlot :: Int -- ^ Number of diagonal pixels. Only affects quality, not size.
           -> (Double -> Double -> PixelRGBA8)
              -- ^ Angle and radius in radians and percent respectively.
           -> Tree
circlePlot density fn =
    scaleToHeight screenHeight $ flipYAxis $
    embedImage $ generateImage gen density density
  where
    cN = fromIntegral $ density `div` 2 - 1
    gen x y =
      let radius = sqrt ((fromIntegral x-cN)**2 + (fromIntegral y-cN)**2)
          ang = atan2 (fromIntegral y-cN) (fromIntegral x-cN)
      in if radius > cN
        then PixelRGBA8 0 0 0 0
        else fn ang (radius / cN)
