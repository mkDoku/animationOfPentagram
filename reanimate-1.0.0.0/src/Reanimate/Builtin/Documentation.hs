{-|
Module      : Reanimate.Builtin.Documentation
Copyright   : Written by David Himmelstrup
License     : Unlicense
Maintainer  : lemmih@gmail.com
Stability   : experimental
Portability : POSIX

This module contains convenience functions used in documention
GIFs for a consistent look and feel.

-}
module Reanimate.Builtin.Documentation where

import Reanimate.Animation
import Reanimate.Svg
import Reanimate.Raster
import Reanimate.Constants
import Codec.Picture

-- | Default environment for API documentation GIFs.
docEnv :: Animation -> Animation
docEnv = mapA $ \svg -> mkGroup
  [ mkBackground "white"
  , withFillOpacity 0 $
    withStrokeWidth 0.1 $
    withStrokeColor "black" (mkGroup [svg]) ]

-- | <<docs/gifs/doc_drawBox.gif>>
drawBox :: Animation
drawBox = mkAnimation 2 $ \t ->
  partialSvg t $ pathify $
  mkRect (screenWidth/2) (screenHeight/2)

-- | <<docs/gifs/doc_drawCircle.gif>>
drawCircle :: Animation
drawCircle = mkAnimation 2 $ \t ->
  partialSvg t $ pathify $
  mkCircle (screenHeight/3)

-- | <<docs/gifs/doc_drawProgress.gif>>
drawProgress :: Animation
drawProgress = mkAnimation 2 $ \t ->
  mkGroup
  [ mkLine (-screenWidth/2*widthP,0)
           (screenWidth/2*widthP,0)
  , translate (-screenWidth/2*widthP + screenWidth*widthP*t) 0 $
    withFillOpacity 1 $ mkCircle 0.5 ]
  where
    widthP = 0.8

-- | Render a full-screen view of a color-map.
showColorMap :: (Double -> PixelRGB8) -> SVG
showColorMap f = center $ scaleToSize screenWidth screenHeight $ embedImage img
  where
    width = 256
    height = 1
    img = generateImage pixelRenderer width height
    pixelRenderer x _y = f (fromIntegral x / fromIntegral (width-1))

-- | Default background color for videos on reanimate.rtfd.io
rtfdBackgroundColor :: PixelRGBA8
rtfdBackgroundColor = PixelRGBA8 252 252 252 0xFF
