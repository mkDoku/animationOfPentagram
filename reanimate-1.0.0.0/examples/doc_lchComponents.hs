#!/usr/bin/env stack
-- stack runghc --package reanimate
module Main(main) where

import Reanimate
import Reanimate.Builtin.Documentation
import Reanimate.ColorComponents
import Codec.Picture

main :: IO ()
main = reanimate $ docEnv $ animate $ const $ showColorMap $ interpolateRGB8 lchComponents yellow blue
  where
    yellow = PixelRGB8 0xFF 0xFF 0x00
    blue = PixelRGB8 0x00 0x00 0xFF
