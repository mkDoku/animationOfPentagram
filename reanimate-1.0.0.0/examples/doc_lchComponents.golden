0<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
1<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
2<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
3<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
4<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
5<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
6<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
7<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
8<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
9<svg viewBox="-8 -4.5 16 9" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" preserveAspectRatio="none"><g stroke-width="0.05" transform="scale(1, -1)"><g><g stroke-width="0" fill="#FFFFFF" transform="translate(-8, -4.5)" fill-opacity="1"><rect width="16" height="9" /></g><g stroke-width="0.1" stroke="#000000" fill-opacity="0"><g transform="translate(0, 0)"><g transform="scale(0.0625, 9)"><g transform="scale(1, -1)"><image width="256" height="1" x="-128" y="-0.5" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAABCAIAAAC+O+cgAAACVklEQVR4nB3CCUzNAQDH8e/MZrM1m1lma2bN1pqZWcMMa67F3GfuY7TEEpMjEZYjhMpVodyUJEkkISlJlFJJkujSffd6vff/mffZBwmLMIku0S6aRYOoE1WiUpSL7wbFBoUGeQa5Rr9so3+mMSDdGJhmtUuxDkq2Dk60Dom3Do21DrtrcbhpGR5lGRFhcbzYNzK0zym4zzmob1SgeXSAeYyfeayv2cWnd5xX74TNvRPXmia5m6YsNLnO6pnq2jN9XPdM5243++7ZdM2hax6dC+hcROcSOpbR4U77StpX07aWtvW0baR1E60etHjS4kXzNpq9afKhaSeNu2jcTcNe6v2o9+fvQf4eou4IdYHUHqP2BDUnqQ6m+ixVIVSF8ecCvy/xO5zKSCqv8iuaihtU3OLnHcrvUR7DjzjK4ilL4HsipUmUJvMthW+plKRR/JqidIoy+JpFYTaFORTk8uUzX/LJLyCviLwSPpfyqYxP5eRW8LGSnCpyavhQR3Y92Y28byarlcx2Mjt5102GiQwzby2kG7zR/6/FK5Fm+1Kkihe2KeK5eGabLJ6KJPHENlE8Fgm2j0S8eCjibB+IWBEj7ot7tnfFHXFb3BI3xQ1x3TZaRIlr4qq4IiJFhAgXl8UlcVFcEOdFmAgVIeKcOCvOiGBxWpwSQeKEOC6OiaMiUBwRh8UhESAOCH+xX+wTe8Ue4St2iZ1ih9guvMU2sVVsEZ7CQ2wSG8UGsU6sEavECrFcLBWLxUIxX8wVs4WbmCGmCVcxWUwU44WLGCNGCSfhKByEvbAT6B8DrYx5Ykn+OwAAAABJRU5ErkJggg==" /></g></g></g></g></g></g></svg>
